### PR TITLE
fix: handle backticks in link text for emphasis processing

### LINKED ISSUE #3777: Link backtick processing seems subtly broken
**Marked version:**

16.3.0

**Describe the bug**

When given a string like this...

```md
**Really weird edge case: bold around what looks like it might be a link, but is actually a link-looking thing with a code specifier in. [Like `this](https://github.com)`. It's now impossible to close the bold**. <https://git`hub.com>
```

...`marked` seems to correctly parse that the original linked-looking thing is actually a codespan. However, at some other stage, it also decides that the backtick _after_ the link is part of another codespan connecting to the one in the middle of the autolink, which prevents closing the `strong` element. 

**To Reproduce**

1. [`marked`](https://marked.js.org/demo/?text=**Really%20weird%20edge%20case%3A%20bold%20around%20what%20looks%20like%20it%20might%20be%20a%20link%2C%20but%20is%20actually%20a%20link-looking%20thing%20with%20a%20code%20specifier%20in.%20%5BLike%20%60this%5D(https%3A%2F%2Fgithub.com)%60.%20It%27s%20now%20impossible%20to%20close%20the%20bold**.%20%3Chttps%3A%2F%2Fgit%60hub.com%3E&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22hooks%22%3A%20null%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%0A%7D&version=16.3.0)
2. [`CommonMark`](https://spec.commonmark.org/dingus/?text=**Really%20weird%20edge%20case%3A%20bold%20around%20what%20looks%20like%20it%20might%20be%20a%20link%2C%20but%20is%20actually%20a%20link-looking%20thing%20with%20a%20code%20specifier%20in.%20%5BLike%20%60this%5D(https%3A%2F%2Fgithub.com)%60.%20It%27s%20now%20impossible%20to%20close%20the%20bold**.%20%3Chttps%3A%2F%2Fgit%60hub.com%3E)

**Expected behavior**

This should be parsed as a `strong` element with a `codespan` inside, followed by an autolink. Right now, it's parsed as regular text including a `codespan` element.

### PR DESCRIPTION
**Pull request by sajdakabir**
[PR] fix: handle backticks in link-like bracket text during emphasis parsing

**Marked version:** 16.3.0  
**Markdown flavor:** CommonMark | GitHub Flavored Markdown

## Description
- Fixes #3777 by updating the `blockSkip` regex to correctly mask backticks inside link-like bracket text before emphasis processing.

## Expectation
Bold/italic should close correctly even when the text includes bracketed segments that look like links and contain code spans:

```markdown
**Really weird edge case: bold around what looks like it might be a link, but is actually a link-looking thing with a code specifier in. [Like `this](https://github.com)`. It's now impossible to close the bold**
```

Expected output:
```html
<p><strong>Really weird edge case: bold around what looks like it might be a link, but is actually a link-looking thing with a code specifier in. [Like <code>this](https://github.com)</code>. It's now impossible to close the bold</strong></p>
```

## Result (before fix)
Emphasis remained unclosed because backticks within the bracketed segment were not masked, interfering with strong/em parsing.

## Technical summary
- The `blockSkip` regex (used to mask links/code/HTML before emphasis) didn’t handle backticks inside the bracket text `[...]`.
- Updated pattern for link text to:
  - Exclude backticks from the generic char class.
  - Explicitly match inline code spans within brackets.
  - Simplify escape handling with `\\.`.

New `blockSkip`:
```
\[(?:[^\[\]\\`]|\\.|`[^`]*`)*?\]\((?:\\.|[^\\\(\)]|\((?:\\.|[^\\\(\)])*\))*\)|`[^`]*?`|<(?!)[^<>]*?>
```

## Contributor
- [x] Tests: existing suite passes (1705 tests), including:
  - `angle_brackets` (ensures `<` and `>` don’t break emphasis)
  - `link_tick_redos` (guards against ReDOS)

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

**Comment by sajdakabir** [comment]
Screenshot 2025-10-05 at 2 20 59 PM
 @UziTech lmk if any changes are needed.

**Review by UziTech — CHANGES_REQUESTED** [review]
Can you remove the indentation changes?

**Comment by sajdakabir** [comment]
@UziTech its done 

> Can you remove the indentation changes?

@UziTech its done

**Review by UziTech — APPROVED** [review]
Thanks 💯

File: test/specs/new/bold_codespan_autolink.html [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<p><strong>Really weird edge case: bold around what looks like it might be a link, but is actually a link-looking thing with a code specifier in. [Like <code>this](https://github.com)</code>. It&#39;s now impossible to close the bold</strong>.</p>
+<p><strong>Bold text with [unmatched `backtick](url) should close</strong>.</p>
```
Comment (styfle) [inline test/specs/new/bold_codespan_autolink.html]: @UziTech I noticed this doesn't match the commonmark output.

Are we okay with that?

[link to dingus](https://spec.commonmark.org/dingus/?text=**Really%20weird%20edge%20case%3A%20bold%20around%20what%20looks%20like%20it%20might%20be%20a%20link%2C%20but%20is%20actually%20a%20link-looking%20thing%20with%20a%20code%20specifier%20in.%20%5BLike%20%60this%5D(https%3A%2F%2Fgithub.com)%60.%20It%27s%20now%20impossible%20to%20close%20the%20bold**.%0A%0A**Bold%20text%20with%20%5Bunmatched%20%60backtick%5D(url)%20should%20close**.)

Comment (sajdakabir) [reply test/specs/new/bold_codespan_autolink.html]: @styfle thanks! I’ve updated the test — this should be resolved now.

Comment (UziTech) [reply test/specs/new/bold_codespan_autolink.html]: I think @styfle was talking about the second paragraph. That is a different bug. Can we just remove the second paragraph from the test?

The first paragraph isn't actually testing this fix currently

Comment (sajdakabir) [reply test/specs/new/bold_codespan_autolink.html]: @UziTech so should i remove the second paragraph?

Comment (UziTech) [reply test/specs/new/bold_codespan_autolink.html]: ya and undo the last fix

Comment (sajdakabir) [reply test/specs/new/bold_codespan_autolink.html]: @UziTech its done , let me know if any other changes are needed.

**Comment by UziTech** [comment]
Looks like this was actually fixed in the latest version by #3785
